core/amber/src/main/scala/edu/uci/ics/amber/clustering/ClusterListener.scala

```diff
@@ -53,9 +54,7 @@ class ClusterListener extends Actor with ActorLogging {
         WorkflowService.getAllWorkflowService.foreach { workflow =>
           val jobService = workflow.jobService.getValue
           if (jobService != null && !jobService.workflow.isCompleted) {
-            try {
-              futures.append(jobService.client.notifyNodeFailure(member.address))
-            }
+            futures.append(jobService.client.notifyNodeFailure(member.address))
```

Comment:
Can we complete the try-catch here instead of removing it? the expected behavior is to shutdown the workflow if it throws an error.

Reply:
I am not sure how to "shutdown" the workflow completely (which involves resource garbage collection I believe). It might be better for you to handle this case since you are more familiar?